src/common/exif.cc

```diff
@@ -1093,103 +1130,116 @@ static bool _exif_decode_exif_data(dt_image_t *img, Exiv2::ExifData &exifData)
         { 1.0206905, 0.0091588, -0.0228796, 0.0115005, 0.9984917, -0.0076762, -0.0043619, 0.0072053,
-          0.8853432 }, // 22 = D75
+          0.8853432 }, // D75
         { 0.8446965, -0.1179225, 0.3948108, -0.1366303, 1.1041226, 0.1291718, 0.0798489, -0.1348999,
-          3.1924009 }, // 17 = Standard light A
+          3.1924009 }, // Standard light A
         { 0.9415037, -0.0321240, 0.0584672, -0.0428238, 1.0250998, 0.0203309, 0.0101511, -0.0161170,
-          1.2847354 }, // 18 = Standard light B
+          1.2847354 }, // Standard light B
         { 0.9904476, -0.0071683, -0.0116156, -0.0123712, 1.0155950, -0.0029282, -0.0035635, 0.0067697,
-          0.9181569 }, // 19 = Standard light C
+          0.9181569 }, //  Standard light C
         { 0.9212269, -0.0449128, 0.1211620, -0.0553723, 1.0277243, 0.0403563, 0.0235086, -0.0391019,
-          1.6390644 }  // F2 = cool white
+          1.6390644 }  // CoolWhiteFluorescent
       };
 
-      if(FIND_EXIF_TAG("Exif.Image.CalibrationIlluminant1")) illu1 = pos->toLong();
-      if(FIND_EXIF_TAG("Exif.Image.CalibrationIlluminant2")) illu2 = pos->toLong();
+      if(FIND_EXIF_TAG("Exif.Image.CalibrationIlluminant1")) illu[0] = (dt_dng_illuminant_t) pos->toLong();
       Exiv2::ExifData::const_iterator cm1_pos = exifData.findKey(Exiv2::ExifKey("Exif.Image.ColorMatrix1"));
-      Exiv2::ExifData::const_iterator cm2_pos = exifData.findKey(Exiv2::ExifKey("Exif.Image.ColorMatrix2"));
-
-      // Which is the wanted colormatrix?
-      // If we have D65 in Illuminant1 we use it; otherwise we prefer Illuminant2 because it's the higher
-      // color temperature and thus closer to D65
-      if(illu1 == 21 && cm1_pos != exifData.end() && cm1_pos->count() == 9 && cm1_pos->size())
+      if((illu[0] != DT_LS_Unknown) && (cm1_pos->count() == 9) && (cm1_pos != exifData.end()))
       {
-        for(int i = 0; i < 9; i++) colmatrix[i] = cm1_pos->toFloat(i);
-        illu = illu1;
+        for(int i = 0; i < 9; i++) colmatrix[0][i] = cm1_pos->toFloat(i);      
       }
-      else if(illu2 != -1 && cm2_pos != exifData.end() && cm2_pos->count() == 9 && cm2_pos->size())
+      else
+        illu[0] = DT_LS_Unknown;
+
+      if(FIND_EXIF_TAG("Exif.Image.CalibrationIlluminant2")) illu[1] = (dt_dng_illuminant_t) pos->toLong();
+      Exiv2::ExifData::const_iterator cm2_pos = exifData.findKey(Exiv2::ExifKey("Exif.Image.ColorMatrix2"));
+      if((illu[1] != DT_LS_Unknown) && (cm2_pos->count() == 9) && (cm2_pos != exifData.end()))
       {
-        for(int i = 0; i < 9; i++) colmatrix[i] = cm2_pos->toFloat(i);
-        illu = illu2;
+        for(int i = 0; i < 9; i++) colmatrix[1][i] = cm2_pos->toFloat(i);      
       }
-      else if(illu1 != -1 && cm1_pos != exifData.end() && cm1_pos->count() == 9 && cm1_pos->size())
+      else
+        illu[1] = DT_LS_Unknown;
+      // So far the Exif.Image.CalibrationIlluminant3 tag and friends have not been implemented and there are no images to test
```

Comment:
Added in Exiv2 0.27.4, but yeah, no samples

Reply:
right